diff --git a/node_modules/@graphprotocol/graph-cli/src/codegen/schema.js b/node_modules/@graphprotocol/graph-cli/src/codegen/schema.js
index af2ff31..82165cb 100644
--- a/node_modules/@graphprotocol/graph-cli/src/codegen/schema.js
+++ b/node_modules/@graphprotocol/graph-cli/src/codegen/schema.js
@@ -127,7 +127,7 @@ module.exports = class SchemaCodeGenerator {
 
 
     let getNonNullable = `return ${typesCodegen.valueToAsc('value', fieldValueType)}`
-    let getNullable = `if (value === null) {
+    let getNullable = `if (value === null || value.kind === ValueKind.NULL) {
                           return null
                         } else {
                           ${getNonNullable}
